src/TrueLayer/Serialization/OneOfJsonConverter.cs

```diff
@@ -33,15 +33,26 @@ public OneOfJsonConverter(OneOfTypeDescriptor descriptor, string discriminatorFi
                 return default;
             }
 
-            if (doc.RootElement.TryGetProperty(_discriminatorFieldName, out var discriminator)
-                && (_descriptor.TypeFactories.TryGetValue(discriminator.GetString()!, out var typeFactory)))
+            doc.RootElement.TryGetProperty(_discriminatorFieldName, out var discriminator);
+            string? discriminatorValue = discriminator.GetString();
+            if (!string.IsNullOrWhiteSpace(discriminatorValue)
+                && (_descriptor.TypeFactories.TryGetValue(discriminatorValue, out var typeFactory)))
             {
                 return InvokeDiscriminatorFactory(options, readerClone, typeFactory);
             }
 
             // Fallback to status field
-            if (doc.RootElement.TryGetProperty("status", out discriminator)
-                && (_descriptor.TypeFactories.TryGetValue(discriminator.GetString()!, out typeFactory)))
+            doc.RootElement.TryGetProperty("status", out discriminator);
+            string? statusValue = discriminator.GetString();
+            if (!string.IsNullOrWhiteSpace(statusValue)
+                && (_descriptor.TypeFactories.TryGetValue(statusValue, out typeFactory)))
+            {
+                return InvokeDiscriminatorFactory(options, readerClone, typeFactory);
+            }
+
+            var statusTypeDiscriminator = string.Join("_", statusValue, discriminatorValue);
+            if (!string.IsNullOrWhiteSpace(statusTypeDiscriminator)
+                && (_descriptor.TypeFactories.TryGetValue(statusTypeDiscriminator, out typeFactory)))
```

Comment:
For the [payout payments](https://docs.truelayer.com/reference/merchant-account-get-transactions#:~:text=EXTERNAL%20PAYMENT-,PENDING,-PAYOUT) resulted from the `GetTransactions` for the `MerchantAccounts`, the discriminator is the concatenation of the `status` and `type` field: `status_type`.

Reply:
for context: On the Java lib we treat them as the same item, just with some optional fields, for simplicity